src/index.tsx

```diff
@@ -25,6 +25,12 @@ type Font = {
 
 type PropertyFunction<T> = () => T;
 
+function partialEnd<R, O>(thisArg: O, callback: (...curryArgs: any[]) => R, ...predefinedArgs: unknown[]) {
+    return function (...args: unknown[]) {
+      return callback.apply(thisArg, [...args, ...predefinedArgs]);
+    };
+}
```

Comment:
Yeah, you are right. I think this is simply because of the naming of the generic types and the name of the function itself. This function is important to the backwards compatibility so one does not need to always pass the event object when calling `handlePrint()` returned from `useReactPrint({ ... })`.

Without currying, this will not work as expected and the next release will have to be a major version number bump release as there will be breaking changes.

I have taken the time to rename and format the function and the generic typing accordingly